### ISSUE TITLE
Support macros in impl blocks

### ISSUE THREAD

**Issue by matklad** [rust-analyzer/rust-analyzer#2459]
While solving advent of code, I've noticed the lack of completion for `u64::saturating_sub`. The problem is that it comes from a macro-generated method:

```rust
#[lang = "i64"]
impl i64 {
    int_impl! { i64, i64, u64, 64, -9223372036854775808, 9223372036854775807, "", "", 12,
         "0xaa00000000006e1", "0x6e10aa", "0x1234567890123456", "0x5634129078563412",
         "0x6a2c48091e6a2c48", "[0x56, 0x34, 0x12, 0x90, 0x78, 0x56, 0x34, 0x12]",
         "[0x12, 0x34, 0x56, 0x78, 0x90, 0x12, 0x34, 0x56]", "", "" }
}
```

We should expand macros in this position, it should be handled here:

https://github.com/rust-analyzer/rust-analyzer/blob/780f476b4f438d473bc2e2299c2b8bf0a6fb9257/crates/ra_hir_def/src/data.rs#L177-L206

**Comment by edwin0cheng** [comment]
It can be closed, right ?

**Comment by matklad** [comment]
do we resolve saturated_sub now?

**Comment by edwin0cheng** [comment]
Nope. It is because `int_impl!` itself used another macro :(

**Comment by kjeremy** [comment]
Wow I just hit this 10 minutes ago.

**Comment by edwin0cheng** [comment]
It should fix by #2626, which show tons of function when I do:
```
fn main() {
    let i = 0u64;
    i.<|>
}
```

**Comment by flodiebold** [comment]
Yeah, as far as I can tell this is fixed.
